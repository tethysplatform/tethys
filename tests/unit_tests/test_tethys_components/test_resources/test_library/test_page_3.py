def test(lib):
    """This comment is here as a test to ensure certain code gets executed"""
    # Register additional packages
    lib.register(
        package="fake-package", accessor="fp", styles=["style1.css", "style2.css"]
    )

    # Event handlers
    def handle_on_ready(e):
        print("Your video is ready.")

    # Layout
    return lib.html.div()(
        lib.rp.ReactPlayer(
            url="https://www.youtube.com/watch?v=xvFZjo5PgG0", onReady=handle_on_ready
        )
    )
