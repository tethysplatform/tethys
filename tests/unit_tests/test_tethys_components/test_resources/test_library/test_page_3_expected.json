{"tagName": "div", "children": [{"tagName": "ReactPlayer", "attributes": {"url": "https://www.youtube.com/watch?v=xvFZjo5PgG0"}, "eventHandlers": {"onReady": null}, "importSource": {"source": "test_page_3.js", "sourceType": "NAME", "fallback": "\u231b", "unmountBeforeUpdate": false}}]}